### PR TITLE
Allow setting gha environment variables

### PR DESCRIPTION
**Pull request by ericof**
To set environment variables to be used by all jobs,
specify the following key:

```toml
[github]
env = """
  debug: 1
  image-name: 'org/image'
  image-tag: 'latest'
"""
```

### CODE REVIEW

**Review by jensens — APPROVED** [review]

File: config/default/meta.yml.j2 [hunk 1]
```diff
@@ -10,6 +10,21 @@ on:
       - main
   workflow_dispatch:
 
+{% if env %}
+env:
+%(env)s
+{% else %}
```
Comment (gforcada) [inline config/default/meta.yml.j2:16]: That's an interesting approach, if the setting is already used, do not add the inline documentation about it... then though, when adding/removing that setting the diff will be bigger as the documentation will be added/removed as well 🤔

**Review by gforcada — CHANGES_REQUESTED** [review]
There's at least a suspicious line deleted that I'm not 100% we want, otherwise looks good.

If you can be extra nice and add a `news/164.feature` news entry, that would be awesome 🤩 🍀

**Review by gforcada — CHANGES_REQUESTED** [review]
There's at least a suspicious line deleted that I'm not 100% we want, otherwise looks good.

If you can be extra nice and add a `news/164.feature` news entry, that would be awesome 🤩 🍀

File: config/config-package.py [hunk 2]
```diff
@@ -383,7 +384,6 @@ def gha_workflows(self):
             destination=destination,
             **options
         )
-        return self.copy_with_meta('meta.yml.j2', destination=destination)
```
[the remark this answers is not in the document]
Comment (ericof) [reply config/config-package.py:386]: @gforcada Look at the previous line :-)
This code was never run

File: config/default/meta.yml.j2 (hunk 1 above)
Comment (ericof) [reply config/default/meta.yml.j2:16]: But we would benefit of having a smaller file :-)

**Pull request merged**